libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -751,16 +830,33 @@ void AP_UAVCAN::handle_actuator_status(AP_UAVCAN* ap_uavcan, uint8_t node_id, co
  */
 void AP_UAVCAN::handle_ESC_status(AP_UAVCAN* ap_uavcan, uint8_t node_id, const ESCStatusCb &cb)
 {
+    const uint8_t esc_index = cb.msg->esc_index;
+    
     // log as CESC message
     AP::logger().Write_ESCStatus(AP_HAL::micros64(),
-                                 cb.msg->esc_index,
+                                 esc_index,
                                  cb.msg->error_count,
                                  cb.msg->voltage,
                                  cb.msg->current,
                                  cb.msg->temperature - C_TO_KELVIN,
                                  cb.msg->rpm,
                                  cb.msg->power_rating_pct);
 
+    WITH_SEMAPHORE(_telem_sem);
+
+    if (esc_index > UAVCAN_SRV_NUMBER) {
+        printf("UAVCAN: invalid esc index: %d. max index allowed: %d\n\r", esc_index, UAVCAN_SRV_NUMBER);
```

Comment:
Leaving this debug in as-is is probably not appropriate; ESC data can come in a rapid rate of knots and this might cause unwanted load.  If you think some output would be appropriate, decimating this message and using gcs().send_text(...) would be more appropriate.

Reply:
Maybe it would be better to use `debug_uavcan(1, "...");` instead? I see this is the way logging works in this file.
As for the load - I agree. we can remove this debug completely or use some flag to print it only once. What do you think should be done?